csrc/device_lower/pass/index.cpp

```diff
@@ -1568,6 +1568,74 @@ static inline DataType getMmaOutType(TensorView* mma_out) {
   return ArrayType{std::make_shared<DataType>(DataType::Float), (size_t)size};
 }
 
+namespace {
+std::pair<Val*, Val*> hardCodedIndexGenerationForStMatrix(
+    const LoadStoreOp* ldst,
+    const int64_t output_m_extent,
+    const int64_t output_n_extent) {
+  NVF_ERROR(
+      (output_m_extent == 8 && output_n_extent == 8) ||
+          (output_m_extent == 16 && output_n_extent == 8) ||
+          (output_m_extent == 16 && output_n_extent == 16),
+      "size not currently supported for stmatrix");
+
+  auto num_regs = (output_m_extent) / 8 * (output_n_extent) / 8;
+  auto as_type = ArrayType{
+      std::make_shared<DataType>(DataType::UInt32),
+      static_cast<size_t>(num_regs)};
+
+  Val* in = IrBuilder::create<kir::TensorIndex>(
+      dynamic_cast<TensorView*>(ldst->in()),
+      IrBuilder::create<Val>(0, DataType::Index),
+      as_type);
+
+  Val* out_index;
+  // This will hanlde 8x8 and 16x8.
+  if (output_n_extent == 8) {
+    // T_shared[toSmem(T_shared) + 16 * tidx.x]
+    out_index = IrBuilder::addExpr(
+        IrBuilder::baseAddressExpr(dynamic_cast<TensorView*>(ldst->out())),
+        IrBuilder::mulExpr(
+            IrBuilder::create<Val>(16, DataType::Index),
+            IrBuilder::create<NamedScalar>("threadIdx.x", DataType::Index)));
+  } else if (output_n_extent == 16) {
+    // This will hanlde 16x16
+    // T_shared[toSmem(T_shared) + 256 * (tidx.x / 16) +  32 * (tidx.x%8)  +
+    // ((tidx.x/8)%2) * 16]
```

Comment:
Looks like you are doing:
```
01
23
```
but it actually needs to be
```
02
13
```
according to https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#wgmma-64n16-d

Reply:
Hmm, interesting.
I tried out:
02
13
Which gave me incorrect results (I then went with how LdMatrix does it). Let me try it out again.

Reply:
You were right, this should have been:
02
13. 

The reason it worked previously was because I scheduled input as operand B.